flaml/searcher/blendsearch.py

```diff
@@ -637,4 +702,31 @@ def _select_thread(self) -> Tuple:
     def _create_condition(self, result: Dict) -> bool:
         ''' create thread condition
         '''
-        return len(self._search_thread_pool) < 2
+        if self._points_to_evaluate:
+            # still evaluating user-specified init points
+            return False
+        if len(self._search_thread_pool) == 2:
+            return False
+        if self._candidate_start_points and self._thread_count == 1:
+            # result needs to match or exceed the best candidate start point
+            obj_best = min(
+                self._ls.metric_op * r[self._ls.metric]
+                for r in self._candidate_start_points.values() if r)
+            return result[self._ls.metric] * self._ls.metric_op <= obj_best
+        else:
+            return True
+
+    def on_trial_complete(self, trial_id: str, result: Optional[Dict] = None,
+                          error: bool = False):
+        super().on_trial_complete(trial_id, result, error)
+        if self._candidate_start_points \
+           and trial_id in self._candidate_start_points:
+            # the trial is a candidate start point
+            if len(self._search_thread_pool) == 2:
+                # ls thread created from this candidate start point
+                del self._candidate_start_points[trial_id]

```

Comment:
Can you move this line to the place where ls thread is created from the candidate start point, like after line 302.